### PR TITLE
clientudp: parse source header

### LINKED ISSUE #107: Parse source header from SETUP response (if present)
@aler9 wrote:
> Hello, in order to load balance i usually prefer the second configuration you described (the 302 redirect-way), since the load balancer can be shut down or restarted independently from the stream, and it's the standard way to perform the task (for instance, it is used in Eclipse Leshan).

> Anyway it's possible to enhance the client to support your configuration, but in my opinion this can be done not by disabling the IP check, but by editing the SETUP response sent by the server and adding a source parameter, that contains the source IP of the UDP packets (like the RFC tells to do):

```
RTSP/1.0 200 OK
CSeq: 302
Date: 23 Jan 1997 15:35:06 GMT
Session: 47112344
Transport: RTP/AVP;unicast;
client_port=4588-4589;server_port=6256-6257,source=192.168.0.13
```

The parsing would need to go i.e. here https://github.com/aler9/gortsplib/blob/47caa2ed2cb8b6e60ea7b5930e09f04ef5592488/pkg/headers/transport.go#L143

and used in this conditional:
https://github.com/aler9/gortsplib/blob/47caa2ed2cb8b6e60ea7b5930e09f04ef5592488/clientudpl.go#L168

_Originally posted by @aler9 in https://github.com/aler9/gortsplib/issues/58#issuecomment-983045552_

### PR DESCRIPTION
**Pull request by tmatth**
Fixes #107

### CODE REVIEW

File: clientudpl.go [hunk 1]
```diff
@@ -187,7 +187,9 @@ func (u *clientUDPListener) processPlayRTP(now time.Time, payload []byte) {
 	pkt.Header.Padding = false
 	pkt.PaddingSize = 0
 
-	u.c.tracks[u.trackID].rtcpReceiver.ProcessPacketRTP(now, pkt)
+	if u.c.tracks[u.trackID].rtcpReceiver != nil {
+		u.c.tracks[u.trackID].rtcpReceiver.ProcessPacketRTP(now, pkt)
+	}
```
Comment (tmatth) [inline clientudpl.go:192]: @aler9 I can split this into its own PR, I just hit this issue in testing.

**Comment by tmatth** [comment]
This needs to handle hostnames (e.g., localhost) to pass CI.

File: clientudpl.go (hunk 1 above)
Comment (aler9) [reply clientudpl.go:192]: this has been fixed in d1cd635 , thanks

**Comment by tmatth** [comment]
> This needs to handle hostnames (e.g., localhost) to pass CI.

@aler9 I've added a commit which falls back to `net.LookupHost` if the source field fails to parse as an IP. This can be squashed or dropped (the RFC isn't really clear about whether hostnames are allowed but I figured since you were using `localhost` in the tests already it made sense to support it).

**Comment by aler9** [comment]
I added tests and merged, thank you very much. This will surely make easier to communicate with servers hosted behind a NAT.

**Pull request merged**